payments/models.py

```diff
@@ -57,52 +66,102 @@ def external_account(self, bank_account):
 
 class StripeEvent(models.Model):
     event_id = models.CharField(primary_key=True, max_length=100, blank=True)
+    user_id = models.CharField(max_length=100, blank=True)
     type = models.CharField(max_length=100, blank=True)
     message_text = models.TextField()
+    verified = models.BooleanField(default=False)
     processed = models.BooleanField(default=False)
+    created = models.DateTimeField(null=True, blank=True)
 
+    @property
     def message(self):
         return json.loads(self.message_text)
 
     def save(self, *args, **kwargs):
+        self.created = timezone.now()
+        super(StripeEvent, self).save(*args, **kwargs)
+
         try:
-            webhooks[self.type].process(self)
+            self.user_id = self.message_text['user_id']
+            retrieved_event = stripe.Event.retrieve(
+                id=self.event_id, stripe_account=self.user_id)
         except KeyError:
+            retrieved_event = stripe.Event.retrieve(id=self.event_id)
+
+        if retrieved_event:
+            self.verified = True
+            self.type = retrieved_event['type']
+            self.message_text = json.dumps(retrieved_event)
+        super(StripeEvent, self).save(*args, **kwargs)
```

Comment:
This `super.save()` should move into the `if` block, so we only save if there really was an event from Stripe. And we should add an `else` block which bails out and/or logs a message like, "Could not retrieve Stripe Event ID: <id>"